src/runtime/pkg/containerd-shim-v2/container.go

```diff
@@ -26,19 +27,48 @@ type container struct {
 	exitIOch    chan struct{}
 	stdinPipe   io.WriteCloser
 	stdinCloser chan struct{}
-	exitCh      chan uint32
+	exitCh      chan containerExit
 	id          string
 	stdin       string
 	stdout      string
 	stderr      string
 	bundle      string
 	cType       vc.ContainerType
 	exit        uint32
-	status      task.Status
+	status      containerStatusMutex
 	terminal    bool
 	mounted     bool
 }
 
+type containerStatusMutex struct {
+	_mutex  sync.Mutex
```

Comment:
Why use a `_` prefix for the fields?

Reply:
`¯\_(ツ)_/¯`

Reply:
I wanted to use `_` prefix because I wanted the struct members where private, and only accessed from the methods, specially because it has to be locked . I have moved to its own package